optuna/integration/skopt.py

```diff
@@ -0,0 +1,260 @@
+                    y = trial.suggest_int('y', 0, 10)
+                    return x**2 + y
+
+                sampler = optuna.integration.SkoptSampler()
+                study = optuna.create_study(sampler=sampler)
+                study.optimize(objective, n_trials=100)
+
+    Args:
+        independent_sampler:
+            A :class:`~optuna.samplers.BaseSampler` instance that is used for independent
+            sampling. The parameters not contained in the relative search space are sampled
+            by this sampler.
+            The search space for :class:`~optuna.integration.SkoptSampler` is determined by
+            :func:`~optuna.samplers.product_search_space()`.
+
+            If :obj:`None` is specified, :class:`~optuna.samplers.RandomSampler` is used
+            as the default.
+
+            .. seealso::
+                :class:`optuna.samplers` module provides built-in independent samplers
+                such as :class:`~optuna.samplers.RandomSampler` and
+                :class:`~optuna.samplers.TPESampler`.
+
+        warn_independent_sampling:
+            If this is :obj:`True`, a warning message is emitted when
+            the value of a parameter is sampled by using an independent sampler.
+
+            Note that the parameters of the first trial in a study are always sampled
+            via an independent sampler, so no warning messages are emitted in this case.
+        skopt_kwargs:
+            Keyword arguments passed to the constructor of
+            `skopt.Optimizer <https://scikit-optimize.github.io/#skopt.Optimizer>`_
+            class.
+
+            Note that ``dimensions`` argument in ``skopt_kwargs`` will be ignored
+            because it is added by :class:`~optuna.integration.SkoptSampler` automatically.
+
+    """
+
+    def __init__(self, independent_sampler=None, warn_independent_sampling=True,
+                 skopt_kwargs=None):
+        # type: (Optional[BaseSampler], bool, Optional[Dict[str, Any]]) -> None
+
+        _check_skopt_availability()
+
+        self._skopt_kwargs = skopt_kwargs or {}
+        if 'dimensions' in self._skopt_kwargs:
+            del self._skopt_kwargs['dimensions']
+
+        self._independent_sampler = independent_sampler or samplers.RandomSampler()
+        self._warn_independent_sampling = warn_independent_sampling
+
+    def infer_relative_search_space(self, study, trial):
+        # type: (InTrialStudy, FrozenTrial) -> Dict[str, BaseDistribution]
+
+        search_space = {}
+        for name, distribution in samplers.product_search_space(study).items():
+            if distribution.single():
+                if not isinstance(distribution, distributions.CategoricalDistribution):
+                    # `skopt` cannot handle non-categorical distributions that contain just
+                    # a single value, so we skip this distribution.
+                    #
+                    # Note that `Trial` takes care of this distribution when suggestion.
```

Comment:
Please rewrite this sentence like: Note that `Trial` takes care of this distribution during suggestion.

Reply:
Thanks. Fixed: https://github.com/pfnet/optuna/pull/399/commits/86551d667b0473d5a669d7d4b3c6981d5f5b38ae